framework/Source/Operations/HistogramEqualization.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  HistogramEqualization.swift
+//
+//
+//  Created by Chiaote Ni on 2024/6/4.
+//
+
+import Foundation
+
+public class HistogramEqualization: ImageProcessingOperation {
+    public var maximumInputs: UInt = 1
+    public let targets = TargetContainer()
+    public let sources = SourceContainer()
+
+    private var renderer = HistogramEqualizationRenderer()
+    private let textureInputSemaphore = DispatchSemaphore(value: 1)
+
+    private var inputTexture: Texture?
+
+    public func transmitPreviousImage(to target: any ImageConsumer, atIndex: UInt) {}
+
+    public func newTextureAvailable(_ texture: Texture, fromSourceIndex: UInt) {
+        guard fromSourceIndex == 0 else {
+            assertionFailure("sourceIndex out of range")
+            return
+        }
+
+        let _ = textureInputSemaphore.wait(timeout:DispatchTime.distantFuture)
+        defer {
+            textureInputSemaphore.signal()
+        }
```

Comment:
make sure this is safe

Reply:
Hi @yyjim I'm going to merge this PR first, and I'll check this in PR: 
[fix: dead lock from process image](https://github.com/cardinalblue/GPUImage3/pull/20)